### PR TITLE
[Nova] Move credentials into Secret objects

### PR DESCRIPTION
**Pull request by joker-at-work**
Please see the individual commits' messages.

### CODE REVIEW

**Comment by joker-at-work** [comment]
Tested in a lab to create a VM, attach a volume, look at its console. migration tasks worked and `nova-bigvm` didn't spew any errors either.

**Review by grandchild — CHANGES_REQUESTED** [review]

**Comment by grandchild** [comment]
i can't say that i'm confident that i have understood everything...

File: openstack/nova/templates/bin/_db-migrate.tpl [hunk 1]
```diff
@@ -39,6 +39,28 @@ update_cell() {
...
 
 
+get_connection_from_file() {
+	file="${1}"
+	python3 <<-EOF
+		from configparser import ConfigParser
+		c = ConfigParser()
+		c.read("${file}")
+		print(c['database']['connection'])
+	EOF
```
[the remark this answers is not in the document]
Comment (joker-at-work) [reply openstack/nova/templates/bin/_db-migrate.tpl:49]: Yes, the whole patch worked for me including the migration task.

Did you maybe replace the tab characters with spaces?

**Comment by joker-at-work** [comment]
> i can't say that i'm confident that i have understood everything...

any questions that would help in understanding (parts) better?

File: openstack/nova/templates/bin/_db-migrate.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (grandchild) [reply openstack/nova/templates/bin/_db-migrate.tpl:49]: yeah, i did. i did not know the _type_ of whitespace makes this behave differently. 🤯 😨 🤦🏾

**Review by grandchild — APPROVED** [review]

**Pull request merged**